--- a/openbsd-compat/openbsd-compat.h	2025-02-18 16:15:08.000000000 +0800
+++ b/openbsd-compat/openbsd-compat.h	2026-07-17 04:11:43.807373890 +0800
@@ -270,9 +270,20 @@
 long long strtonum(const char *, long long, long long, const char **);
 #endif
 
+
+/* ANDROID_GETPWENT: Bionic only declares getpwent family for API >= 26 */
+#if defined(__ANDROID__) && defined(__ANDROID_API__) && (__ANDROID_API__ < 26)
+struct passwd;
+struct passwd *getpwent(void);
+void setpwent(void);
+void endpwent(void);
+#endif
+
 /* multibyte character support */
 #ifndef HAVE_MBLEN
-# define mblen(x, y)	(1)
+# ifndef __ANDROID__
+#  define mblen(x, y)	(1)
+# endif
 #endif
 
 #ifndef HAVE_WCWIDTH
